lib/SIL/IR/SILDeclRef.cpp

```diff
@@ -221,6 +228,12 @@ bool SILDeclRef::isImplicit() const {
 }
 
 SILLinkage SILDeclRef::getLinkage(ForDefinition_t forDefinition) const {
+
+  // Prespecializations are public.
```

Comment:
Only specializations of public symbols though, right?

Reply:
Pre-specialization are intentionally always public.

For example, there is going to be a public entry point for the specialized version of:

```
@_alwaysEmitIntoClient
@_specialize(exported: true, where T == Int)
internal func testGeneric<T>(t: T) {
  print(t)
}
```